main.tf

```diff
@@ -34,12 +34,16 @@ resource "aws_sns_topic_subscription" "this" {
   endpoint               = var.subscribers[each.key].endpoint
   endpoint_auto_confirms = var.subscribers[each.key].endpoint_auto_confirms
   raw_message_delivery   = var.subscribers[each.key].raw_message_delivery
-  redrive_policy = var.sqs_dlq_enabled ? coalesce(var.redrive_policy, jsonencode({
-    deadLetterTargetArn = join("", aws_sqs_queue.dead_letter_queue.*.arn)
-    maxReceiveCount     = var.redrive_policy_max_receiver_count
+  redrive_policy = var.sqs_dlq_enabled ? coalesce(var.redrive_policy, jsonencode(
```

Comment:
Looks like this redrive policy may need to be added directly to the sqs queue

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sqs_queue

Then the redrive policy for sns topic subscription may only support the deadLetterTargetArn

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic_subscription#redrive_policy